crypto/groups/modcyclicgroup.go

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2017 XLAB d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package groups
+
+// TODO: this will replace dlog/zp.go
+
+import (
+	"math/big"
+	"github.com/xlab-si/emmy/crypto/common"
+)
+
+// ModCyclicGroup is a cyclic group in modular arithmetic. It holds P = Q * R + 1 for some R.
+type ModCyclicGroup struct {
+	P               *big.Int // modulus of the group
+	G               *big.Int // generator of subgroup
+	Q 				*big.Int // order of G
+}
+
+// NewSchnorrGroup generates random ModCyclicGroup with generator G and
+// parameters P and Q where P = R * Q + 1 for some R. Order of G is Q.
+func NewSchnorrGroup(qBitLength int) (*ModCyclicGroup, error) {
+	group := ModCyclicGroup{}
+	g, q, p, err := common.GetSchnorrGroup(qBitLength)
```

Comment:
Wouldn't it be better if you move the logic from `common.GetSchnorrGroup` directly to this `NewSchnorrGroup` function (and remove it from there)? I prefer to have all the logic related to schnorr groups inside one crypto package.